lemur/utility.py

```diff
@@ -837,46 +827,32 @@ def delete_class(class_id):
     ds.commit()
 
 
-# change the students in a class with class id
-def change_class_students(class_id, student_names):
-    # clean up all the white space in the string
-    student_names_str = ''.join(student_names.split())
-    # get role objects from the Role table
-    role_student = get_role('Student')
-    # Process the string to strip all the white space;
-    # then split it over commas and join with commas
-    student_names = student_names_str.split(',')
-    # Check the correctness of the string to be strings delimited only by comma
-    pattern_for_student_names = '[a-zA-Z](,[a-zA-Z])*'
-    if (re.match(pattern_for_student_names, student_names_str) is None):
-        err_msg = 'The pattern for student names is not correct'
-        return err_msg
-    student_ids = [generate_user_id(s) for s in student_names]
+# Change the users(both professors and students) in a class
+def change_class_users(class_id, new_users):
+    if not class_exists(class_id):
+        return 'Class with id: {} doesn\'t exist'.format(class_id)
     the_class = get_class(class_id)
-
-    # Delete the students that you chose to be removed from the class
-    # if they are not in any other class
-    # If they are in another class, remove the class from their class list
-    for s in the_class.users:
-        if s.role_name == 'Student' and (not (s.id in student_ids)):
-            if len(s.classes) == 1:
-                ds.delete(s)
-            else:
-                s.classes = [c for c in s.classes if c.id != class_id]
-    ds.commit()
-    # Check the that the students you are adding already exist if they don't,
-    # create the student account and add the class to their class list
-    for s in student_names:
-        s_id = generate_user_id(s)
-        if not user_exists(s_id):
-            ds.add(m.User(id=s_id, username=s,
-                          role=role_student, classes=[the_class],
-                          labs=the_class.labs))
+    old_users = the_class.users
+    the_class.users = []
+    # Add new users to the class;
+    # add the associated labs to these users lab list
+    for u in new_users:
+        if not user_exists(str(u)):
+            ds.rollback()
+            return 'User with username: {} doesn\'t exist'.format(u)
         else:
-            the_classes = get_user(s_id).classes
-            if not (class_id in [c.id for c in the_classes]):
-                get_user(s_id)
+            user = get_user(u)
+            user.labs = the_class.labs
+            the_class.users.append(user)
     ds.commit()
+    # Delete the class and the associated labs from old users
+    for u in old_users:
+        new_lab_list = []
+        for lab in u.labs:
+            if lab.the_class.id != class_id:
+                new_lab_list.append(lab)
+        u.classes = [c for c in u.classes if c.id != class_id]
+        u.labs = new_lab_list
```

Comment:
i think this needs to happen before you commit? Is it actually doing anything as written?